controller/eks-cluster-config-handler.go

```diff
@@ -444,7 +449,16 @@ func (h *Handler) create(config *eksv1.EKSClusterConfig, sess *session.Session,
 
 	displayName := config.Spec.DisplayName
 
-	var err error
+	listOutput, err := eksService.ListClusters(&eks.ListClustersInput{})
+	if err != nil {
+		return config, fmt.Errorf("error listing clusters: %v", err)
+	}
+	for _, cluster := range listOutput.Clusters {
+		if aws.StringValue(cluster) == displayName {
+			return config, fmt.Errorf("cannot create cluster [%s] because a cluster in EKS exists with the same name", displayName)
+		}
+	}
```

Comment:
Can you throw a comment above here or put it in a function with a name describing what it's doing? I think at first glance someone could be confused as to how this and what is done in validateCreate is different.